usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateShapeIdReferenceInStopTimeTest.java

```diff
@@ -0,0 +1,293 @@
+
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Logger mockLogger = mock(Logger.class);
+
+        final ValidateShapeIdReferenceInStopTime underTest =
+                new ValidateShapeIdReferenceInStopTime(mockDataRepo, mockResultRepo, mockLogger);
+
+        underTest.execute();
+
+        verify(mockLogger, times(1)).info(ArgumentMatchers.eq(
+                "Validating rule 'E034 - `shape_id` must be provided and valid when" +
+                        " stop_times.shape_dist_travelled is provided'" + System.lineSeparator()));
+
+        verify(mockDataRepo, times(1)).getStopTimeAll();
+        verify(mockDataRepo, times(1))
+                .getStopTimeByTripId(ArgumentMatchers.eq("trip id"));
+        verify(mockDataRepo, times(1)).getTripById("trip id");
+
+        verify(mockStopTime, times(1)).getShapeDistTraveled();
+        verify(mockTrip, times(1)).getShapeId();
+
+        final ArgumentCaptor<MissingRequiredValueNotice> captor =
+                ArgumentCaptor.forClass(MissingRequiredValueNotice.class);
+
+        verify(mockResultRepo, times(1)).addNotice(captor.capture());
+
+        final List<MissingRequiredValueNotice> noticeList = captor.getAllValues();
+
+        assertEquals("trips.txt", noticeList.get(0).getFilename());
+        assertEquals("shape_id", noticeList.get(0).getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals("trip id", noticeList.get(0).getEntityId());
+
+        verifyNoMoreInteractions(mockDataRepo, mockLogger, mockResultRepo, mockStopTime, mockTrip);
+    }
+
+    // suppress warning regarding ignored result of method since it is not necessary here.
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    void stopTimeWithNonNullShapeDistTravelledPlusValidTripIdAndInvalidShapeIdShouldGenerateNotice() {
+        final StopTime mockStopTime = mock(StopTime.class);
+        when(mockStopTime.getTripId()).thenReturn("trip id");
+        when(mockStopTime.getStopSequence()).thenReturn(2);
+        when(mockStopTime.getShapeDistTraveled()).thenReturn(5f);
+        final Map<String, TreeMap<Integer, StopTime>> mockStopTimeCollection = new HashMap<>();
+        final TreeMap<Integer, StopTime> innerMap = new TreeMap<>();
+        innerMap.put(2, mockStopTime);
+        mockStopTimeCollection.put("trip id", innerMap);
+
+        final Trip mockTrip = mock(Trip.class);
+        when(mockTrip.getShapeId()).thenReturn(null);
+        when(mockTrip.getTripId()).thenReturn("trip id");
+        when(mockTrip.getShapeId()).thenReturn("non existing shape id");
+
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        when(mockDataRepo.getStopTimeAll()).thenReturn(mockStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("trip id")).thenReturn(innerMap);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(null);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(mockTrip);
```

Comment:
Same here, should `when(mockDataRepo.getTripById("trip id")).thenReturn();` implemented twice?

Reply:
Without doubt, some copy-pasting went wrong 😅. Modifying this.